vyper/venom/ir_node_to_venom.py

```diff
@@ -486,334 +373,143 @@ def _convert_ir_bb(ctx, ir, symbols, variables, allocated_variables):
-                    allocated_variables[var.name] = bb.append_instruction(
-                        "alloca", var.size, var.pos
-                    )
-
-                offset = int(sym_ir.value) - var.pos
-                if offset > 0:
-                    ptr_var = bb.append_instruction("add", var.pos, offset)
-                else:
-                    ptr_var = allocated_variables[var.name]
+        if isinstance(arg_1, IRVariable):
+            symbols[f"&{arg_0.value}"] = arg_1
 
-                bb.append_instruction("mstore", arg_1, ptr_var)
-            else:
-                if isinstance(sym_ir, IRLiteral):
-                    new_var = bb.append_instruction("store", arg_1)
-                    symbols[f"&{sym_ir.value}"] = new_var
-                    # if allocated_variables.get(var.name, None) is None:
-                    allocated_variables[var.name] = new_var
-                return new_var
-        else:
-            if not isinstance(sym_ir, IRLiteral):
-                bb.append_instruction("mstore", arg_1, sym_ir)
-                return None
-
-            sym = symbols.get(f"&{sym_ir.value}", None)
-            if sym is None:
-                bb.append_instruction("mstore", arg_1, sym_ir)
-                if arg_1 and not isinstance(sym_ir, IRLiteral):
-                    symbols[f"&{sym_ir.value}"] = arg_1
-                return None
-
-            if isinstance(sym_ir, IRLiteral):
-                bb.append_instruction("mstore", arg_1, sym)
-                return None
-            else:
-                symbols[sym_ir.value] = arg_1
-                return arg_1
+        ctx.get_basic_block().append_instruction("mstore", arg_1, arg_0)
     elif ir.value == "ceil32":
         x = ir.args[0]
         expanded = IRnode.from_list(["and", ["add", x, 31], ["not", 31]])
-        return _convert_ir_bb(ctx, expanded, symbols, variables, allocated_variables)
+        return _convert_ir_bb(ctx, expanded, symbols)
     elif ir.value == "select":
-        # b ^ ((a ^ b) * cond) where cond is 1 or 0
         cond, a, b = ir.args
-        expanded = IRnode.from_list(["xor", b, ["mul", cond, ["xor", a, b]]])
-        return _convert_ir_bb(ctx, expanded, symbols, variables, allocated_variables)
-
-    elif ir.value in ["sload", "iload"]:
-        arg_0 = _convert_ir_bb(ctx, ir.args[0], symbols, variables, allocated_variables)
-        return ctx.get_basic_block().append_instruction(ir.value, arg_0)
-    elif ir.value in ["sstore", "istore"]:
-        arg_0, arg_1 = _convert_ir_bb_list(ctx, ir.args, symbols, variables, allocated_variables)
-        ctx.get_basic_block().append_instruction(ir.value, arg_1, arg_0)
-    elif ir.value == "unique_symbol":
-        sym = ir.args[0]
-        new_var = ctx.get_next_variable()
-        symbols[f"&{sym.value}"] = new_var
-        return new_var
+        expanded = IRnode.from_list(["with", "b", b, ["xor", "b", ["mul", cond, ["xor", a, "b"]]]])
```

Comment:
order of eval -- this should really be `["with", "cond", cond, ["with", "a", a, ["with", "b" b, ["xor" ... ]]]]`